hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -932,11 +932,15 @@ func (r *HostedClusterReconciler) reconcileCAPIManager(ctx context.Context, hclu
 	}
 
 	// Reconcile CAPI manager deployment
+	capiImage := imageCAPI
+	if _, ok := hcluster.Annotations[hyperv1.ClusterAPIManagerImage]; ok {
+		capiImage = hcluster.Annotations[hyperv1.ClusterAPIManagerImage]
+	}
 	capiManagerDeployment := clusterapi.ClusterAPIManagerDeployment(controlPlaneNamespace.Name)
 	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, capiManagerDeployment, func() error {
 		// TODO (alberto): This image builds from https://github.com/kubernetes-sigs/cluster-api/pull/4709
 		// We need to build from main branch and push to quay.io/hypershift once this is merged or otherwise enable webhooks.
-		return reconcileCAPIManagerDeployment(capiManagerDeployment, capiManagerServiceAccount)
+		return reconcileCAPIManagerDeployment(capiManagerDeployment, capiManagerServiceAccount, capiImage)
```

Comment:
just a thought there's really no need for the image to be a free parameter in the func signature right? the chunk of code reading the annotation could be just part of `reconcileCAPIManagerDeployment`? same thought for the autoscaler

Reply:
imho I think having the image as an explicit parameter makes the reconcile function easier to unit test

Reply:
you are right: conceptually how I sort of believed the refactor of these reconcile functions were done though was they want to limit the amount of data passed in (aka to do what you want we would have to pass in the hostedcluster struct.

I'm open to either approach I just assumed the typical flow was to detect the image outside the reconcile and then pass that in as an explicit parameter

Reply:
>imho I think having the image as an explicit parameter makes the reconcile function easier to unit test

I don't necessarily agree in this case since this is not a dependency that needs to be injected. There's only two different opinionated values the function should support and that unit test should cover: the original one and the one in the annotation and for that is irrelevant the image name is part of the signature. I'm fine either way anyways :)